core/src/main/scala/cats/syntax/applicativeError.scala

```diff
@@ -83,6 +86,9 @@ final class ApplicativeErrorOps[F[_], E, A](private val fa: F[A]) extends AnyVal
   def attempt(implicit F: ApplicativeError[F, E]): F[Either[E, A]] =
     F.attempt(fa)
 
+  def attemptCase[EE](implicit F: ApplicativeError[F, E], tag: ClassTag[EE], ev: EE <:< E): F[Either[EE, A]] =
+    F.recover(F.map(fa)(_.asRight[EE])) { case e: EE => e.asLeft[A] }
```

Comment:
Given that extra inter-dependencies could lead to circular dependencies, perhaps we can by-pass the `either` syntax and use this not much longer alternative: 
  ```suggestion
      F.recover(F.map(fa)(x => Right[EE, A](x))) { case e: EE => Left[EE, A](e) }
```